include/main.hpp

```diff
@@ -2,8 +2,115 @@
 
 #include <chrono>
 #include <functional>
+#include <map>
 #include <memory>
+#include <nlohmann/json.hpp>
+#include <stdexcept>
 #include <string>
 
+#include "common_package/commands.hpp"
 #include "common_package/common_node.hpp"
+#include "common_package/topic_names.hpp"
 #include "rclcpp/rclcpp.hpp"
+#include "structs.hpp"
+
+// Message includes
+#include "interfaces/msg/control.hpp"
+#include "interfaces/msg/gps_position.hpp"
+#include "interfaces/msg/job_finished.hpp"
+#include "interfaces/msg/mission_progress.hpp"
+#include "interfaces/msg/uav_waypoint_command.hpp"
+#include "interfaces/msg/waypoint.hpp"
+
+typedef enum NodeState {
+    init,
+    pre_wait_time,
+    reach_cruise_height,
+    fly_to_waypoint,
+    reach_target_height,
+    post_wait_time,
+} NodeState_t;
+
+/**
+ * @brief A class for the waypoint node.
+ */
+class WaypointNode : public common_lib::CommonNode {
+   private:
+    NodeState_t node_state = init;  //!< Current state of the node
+
+    Command cmd;  //!< Command that will be executed when active
+
+    bool state_first_loop =
+        true;  //!< If set to true, the first event loop after activating the
+               //!< node is happening. Will be set to false when calling
+               //!< get_state_first_loop().
+
+    Position pos;  //!< Current position
+
+    static constexpr uint16_t max_position_msg_time_difference_ms =
+        100;  //!< Maximum age of a position message from FCC bridge
+
+    static constexpr uint16_t max_progress_msg_time_difference_ms =
```

Comment:
std::chrono constructors are already constexpr.
```suggestion
    static constexpr std::chrono::milliseconds max_progress_msg_time_difference{500};
```

Reply:
Let's focus on the real problems.